addons/calendar/tests/test_event_notifications.py

```diff
@@ -185,3 +186,78 @@ def test_email_alarm(self):
                 'subtype': 'mail.mt_note',
             }):
                 self.env['calendar.alarm_manager'].with_context(lastcall=now - relativedelta(minutes=15))._send_reminder()
+
+    def test_email_alarm_recurrence(self):
+        # test that only a single cron trigger is created for recurring events.
+        # Once a notification has been sent, the next one should be created.
+        # It prevent creating hunderds of cron trigger at event creation
+        alarm = self.env['calendar.alarm'].create({
+            'name': 'Alarm',
+            'alarm_type': 'email',
+            'interval': 'minutes',
+            'duration': 1,
+        })
+        cron = self.env.ref('calendar.ir_cron_scheduler_alarm')
+        cron.lastcall = False
+        with self.capture_triggers('calendar.ir_cron_scheduler_alarm') as capt:
+            with freeze_time('2022-04-13 10:00+0000'):
+                now = fields.Datetime.now()
+                self.env['calendar.event'].create({
+                    'name': "Single Doom's day",
+                    'start': now + relativedelta(minutes=15),
+                    'stop': now + relativedelta(minutes=20),
+                    'alarm_ids': [fields.Command.link(alarm.id)],
+                }).with_context(mail_notrack=True)
+                self.env.flush_all()
+                capt.records.ensure_one()
+        with self.capture_triggers('calendar.ir_cron_scheduler_alarm') as capt:
+            with freeze_time('2022-04-13 10:00+0000'):
+                self.env['calendar.event'].create({
+                    'name': "Recurring Doom's day",
+                    'start': now + relativedelta(minutes=15),
+                    'stop': now + relativedelta(minutes=20),
+                    'recurrency': True,
+                    'rrule_type': 'monthly',
+                    'month_by': 'date',
+                    'day': 13,
+                    'count': 5,
+                    'alarm_ids': [fields.Command.link(alarm.id)],
+                }).with_context(mail_notrack=True)
+                self.env.flush_all()
+                self.assertEqual(len(capt.records), 1, "1 trigger should have been created for the whole recurrence")
+                self.env['calendar.alarm_manager']._send_reminder()
+
+            with freeze_time('2022-05-16 10:00+0000'):
+                self.env['calendar.alarm_manager']._send_reminder()
+                self.assertEqual(len(capt.records), 3, "1 more trigger should have been created")
```

Comment:
why 3? is it because we are 2 months after so it created a trigger for 2022-04-13 and 2022-05-16?
was the one for april created by the first `_send_reminder()` or the second?

```suggestion
                self.assertEqual(capt.records.filtered('call_at'), [
                    datetime(...),
                    datetime(...),
                    datetime(...),
                ])
```

Reply:
Hello, thanks for your input ! The code was not correct yet. 
One trigger was created when the calendar.recurrence is created.
Then we run `self.env['calendar.alarm_manager']._send_reminder()` in April (line 228). And then a new ir cron trigger was created with the same call_at value because _send_reminder would force the parameter recurrence_update == true. 

I updated the code to avoid creating multiple ir.cron.trigger with the same call_at value for the same recurrence.
It will not prevent several ir.cron.trigger sharing the same `call_at` for different events/recurrence but it should be a nice improvement.